core/src/main/java/tc/oc/pgm/db/DatastoreImpl.java

```diff
@@ -40,8 +39,7 @@ public DatastoreImpl(File file) throws SQLException {
           "Could not find SQLite3 driver class (likely due to a jar shading issue)", e);
     }
 
-    final Connection connection =
-        DriverManager.getConnection("jdbc:sqlite:" + file.getAbsolutePath());
+    final Connection connection = DriverManager.getConnection(uri);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *connection*. Currently, there are execution paths that do not contain closure statements, e.g., when exception is thrown by Connection.setAutoCommit. Either a) close *connection* in a try-finally block or b) close the resource by declaring *connection* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).